storage/aptosdb/src/pruner/mod.rs

```diff
@@ -95,67 +107,74 @@ impl Pruner {
-            ledger_prune_window: storage_pruner_config
-                .ledger_prune_window
-                .expect("Default prune window must be specified"),
+            state_store_prune_window: storage_pruner_config.state_store_prune_window,
+            ledger_prune_window: storage_pruner_config.ledger_prune_window,
             worker_thread: Some(worker_thread),
             command_sender: Mutex::new(command_sender),
-            min_readable_version: worker_progress_clone,
+            min_readable_versions: worker_progress_clone,
             last_version_sent_to_pruners: Arc::new(Mutex::new(0)),
             pruning_batch_size: storage_pruner_config.pruning_batch_size,
             latest_version: Arc::new(Mutex::new(0)),
         }
     }
 
-    pub fn get_state_store_pruner_window(&self) -> Version {
+    pub fn get_state_store_pruner_window(&self) -> Option<Version> {
         self.state_store_prune_window
     }
 
-    pub fn get_ledger_pruner_window(&self) -> Version {
+    pub fn get_ledger_pruner_window(&self) -> Option<Version> {
         self.ledger_prune_window
     }
 
-    pub fn get_min_readable_version_by_pruner_index(&self, pruner_index: PrunerIndex) -> Version {
-        self.min_readable_version.lock()[pruner_index as usize]
+    pub fn get_min_readable_version_by_pruner_index(
+        &self,
+        pruner_index: PrunerIndex,
+    ) -> Option<Version> {
+        self.min_readable_versions.lock()[pruner_index as usize]
     }
 
-    pub fn get_min_readable_ledger_version(&self) -> Version {
+    pub fn get_min_readable_ledger_version(&self) -> Option<Version> {
         self.get_min_readable_version_by_pruner_index(LedgerPrunerIndex)
     }
     /// Sends pruning command to the worker thread when necessary.
     pub fn maybe_wake_pruner(&self, latest_version: Version) {
         *self.latest_version.lock() = latest_version;
         if latest_version
-            >= *self.last_version_sent_to_pruners.lock() + self.pruning_batch_size as u64
+            >= *self.last_version_sent_to_pruners.as_ref().lock() + self.pruning_batch_size as u64
         {
             self.wake_pruner(latest_version);
-            *self.last_version_sent_to_pruners.lock() = latest_version;
+            *self.last_version_sent_to_pruners.as_ref().lock() = latest_version;
         }
     }
 
     fn wake_pruner(&self, latest_version: Version) {
-        let min_readable_state_store_version =
-            latest_version.saturating_sub(self.state_store_prune_window);
-        let min_readable_ledger_version = latest_version.saturating_sub(self.ledger_prune_window);
+        let mut prune_window_vector = Vec::new();
+        if let Some(state_store_prune_window) = self.state_store_prune_window {
+            prune_window_vector.push(Some(
+                latest_version.saturating_sub(state_store_prune_window),
+            ));
+        } else {
+            prune_window_vector.push(None);
```

Comment:
You can avoid if..else here and write it consicely as 

```
prune_window_vector.push(state_store_prune_window.map(|x| latest_version.saturating_sub(x),
            )

```

Reply:
Done